src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1515,53 +1564,28 @@ protected virtual void AssignImpl(BoundNode node, BoundExpression value, bool is
                 case BoundKind.InlineArrayAccess:
                     {
                         var elementAccess = (BoundInlineArrayAccess)node;
-
-                        // PROTOTYPE(InlineArrays): Ensure adequate test coverage
-
-                        switch (elementAccess.Expression.Kind)
+                        if (written)
                         {
-                            case BoundKind.Local:
-                                {
-                                    var local = (BoundLocal)elementAccess.Expression;
-                                    if (local.LocalSymbol.RefKind != RefKind.None && !isRef)
-                                    {
-                                        // Writing through the (reference) value of a reference local
-                                        // requires us to read the reference itself.
-                                        if (written) VisitRvalue(local, isKnownToBeAnLvalue: true);
-                                    }
-                                    else
-                                    {
-                                        if (written) NoteWrite(local, null, read);
-                                    }
-                                    break;
-                                }
+                            NoteWrite(elementAccess.Expression, value: null, read);
+                        }
 
-                            case BoundKind.InlineArrayAccess:
-                                AssignImpl(elementAccess.Expression, null, isRef, written, read);
+                        if (elementAccess.Expression.Type.HasInlineArrayAttribute(out int length) &&
+                            (elementAccess.Argument.ConstantValueOpt is { SpecialType: SpecialType.System_Int32, Int32Value: 0 } ||
+                             Binder.InferConstantIndexFromSystemIndex(compilation, elementAccess.Argument, length, out _) is 0))
+                        {
+                            int slot = MakeMemberSlot(elementAccess.Expression, elementAccess.Expression.Type.TryGetInlineArrayElementField());
+                            if (slot > 0)
+                            {
+                                SetSlotState(slot, written);
```

Comment:
>SetSlotState(slot, written); [](http://example.com/codeflow?start=32&length=28)

Are we marking the field as definitely assigned if the 0-th element is assigned? If so, is that useful assuming the field presumably represents N contiguous elements, and the other elements have not necessarily been assigned?